test/helm_dependency_example_template_test.go

```diff
@@ -0,0 +1,179 @@
+// We recommend at least 4 cores and 16GB of RAM if you want to run all the tests together.
+
+package test
+
+import (
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	appsv1 "k8s.io/api/apps/v1"
+
+	"github.com/gruntwork-io/terratest/modules/helm"
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	"github.com/gruntwork-io/terratest/modules/logger"
+	"github.com/gruntwork-io/terratest/modules/random"
+)
+
+// This file contains examples of how to use terratest to test helm chart template logic by rendering the templates
+// using `helm template`, and then reading in the rendered templates.
+// There are two tests:
+// - TestHelmBasicExampleTemplateRenderedDeployment: An example of how to read in the rendered object and check the
+//   computed values.
+// - TestHelmBasicExampleTemplateRequiredTemplateArgs: An example of how to check that the required args are indeed
+//   required for the template to render.
+
+// An example of how to verify the rendered template object of a Helm Chart given various inputs.
+func TestHelmDependencyExampleTemplateRenderedDeployment(t *testing.T) {
+	t.Parallel()
+
+	// Path to the helm chart we will test
+	helmChartPath, err := filepath.Abs("../examples/helm-dependency-example")
+	releaseName := "helm-dependency"
+	require.NoError(t, err)
+
+	// Since we aren't deploying any resources, there is no need to setup kubectl authentication or helm home.
+
+	// Set up the namespace; confirm that the template renders the expected value for the namespace.
+	namespaceName := "medieval-" + strings.ToLower(random.UniqueId())
+	logger.Logf(t, "Namespace: %s\n", namespaceName)
+
+	// Setup the args. For this test, we will set the following input values:
+	// - containerImageRepo=nginx
+	// - containerImageTag=1.15.8
+	options := &helm.Options{
+		SetValues: map[string]string{
+			"containerImageRepo":       "nginx",
+			"containerImageTag":        "1.15.8",
+			"basic.containerImageRepo": "nginx",
+			"basic.containerImageTag":  "1.15.8",
+		},
+		KubectlOptions: k8s.NewKubectlOptions("", "", namespaceName),
+	}
+
+	testCases := []struct {
+		name         string
+		templateName string
+	}{
+		{
+			"dependent chart",
+			"templates/deployment.yaml",
+		},
+		{
+			"basic chart",
+			"charts/basic/templates/deployment.yaml",
+		},
+	}
+
+	for _, testCase := range testCases {
+		testCase := testCase
+		t.Run(testCase.name, func(subT *testing.T) {
+			subT.Parallel()
+			// Run RenderTemplate to render the template and capture the output. Note that we use the version without `E`, since
+			// we want to assert that the template renders without any errors.
+			// Additionally, although we know there is only one yaml file in the template, we deliberately path a templateFiles
+			// arg to demonstrate how to select individual templates to render.
+			output := helm.RenderTemplate(t, options, helmChartPath, releaseName, []string{testCase.templateName})
```

Comment:
Not sure if its a good idea to render helm templates in `Parallel()` since it may lead to concurrent dependency resolving in the same directory...